torchvision/models/swin_transformer.py

```diff
@@ -372,14 +372,16 @@ def __init__(
         self.avgpool = nn.AdaptiveAvgPool2d(1)
         self.head = nn.Linear(num_features, num_classes)
 
-        for m in self.modules():
-            if isinstance(m, (nn.Conv2d, nn.Linear)):
-                nn.init.trunc_normal_(m.weight, std=0.02)
-                if m.bias is not None:
-                    nn.init.constant_(m.bias, 0)
-            elif isinstance(m, nn.LayerNorm):
-                nn.init.constant_(m.bias, 0)
-                nn.init.constant_(m.weight, 1.0)
+        self.apply(self._init_weights)
+
+    def _init_weights(self, m): 
```

Comment:
I recommend not to use this idiom as it can lead to unexpected results if the model gets FXed. The model initialization should happen on the constructor until there is proper support for model reinitialization from the nn.Module.